apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx

```diff
@@ -399,6 +399,9 @@ export default async function RootLayout({ children }: { children: React.ReactNo
 </TabPanel>
 </Tabs>
 
+> Remember: With this RootLayout, you will get problems when using static site generation (SSG) somewhere in your project because in "dev" mode, NextJS will complain about the usage of header/cookie methods in the supebase client creation.
```

Comment:
@logemann can you say a bit more about the root layout problems? Just want to help clarify whether the issue is with SSG or more with the header/cookie methods.

cc: @dijonmusters 

```suggestion
> Remember: With this RootLayout, you will get problems when using static site generation (SSG) somewhere in your project because in "dev" mode, NextJS will complain about the usage of header/cookie methods in the Supabase client creation.
```

Reply:
Its related to both. With SSG, you dont have access to http headers because there is no http request at build time. NextJS complains about the usage of next/headers in the RootLayout cause every page in the project relies on this layout, even the SSG pages you put somewhere in your project. 

In supabase shoes, i would rather provide a better auth-helper skeleton project than to put in the comment, but either one of those things should happen. Otherwise you will get tons of inquries about that problem when AppFolder in NextJS is used more widely in the future.